evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -178,15 +178,24 @@ where
                     // because we only need the from, to, value, and data. We can later change this
                     // into 1559, in the cli package, relatively easily once we
                     // know the target chain supports EIP-1559.
-                    self.broadcastable_transactions.push_back(TypedTransaction::Legacy(
-                        TransactionRequest {
-                            from: Some(broadcast.origin),
-                            to: Some(NameOrAddress::Address(call.contract)),
-                            value: Some(call.transfer.value),
-                            data: Some(call.input.clone().into()),
-                            ..Default::default()
-                        },
-                    ));
+                    if !is_static {
```

Comment:
What is a usecase for sending a static function?

If i'm inside a `startBroadcast` context, and I want to make some assertions, I feel like most people's intuitions would be for `staticcalls` to not be broadcasted.  And in the end, they would probably:  close, make the assertion and open the broadcast block again.

Reply:
yeah to be clear i think it makes sense. The crux of the argument, to quote @mds1, is basically: “All calls between start and end broadcast get sent as txs”. Plain and simple. i am pretty open to either though for what its worth

Reply:
> If i'm inside a startBroadcast context, and I want to make some assertions, I feel like most people's intuitions would be for staticcalls to not be broadcasted. And if in the end, they are broadcasted... most people would probably: close, make the assertion and open the broadcast block again. Which just makes everything too crowded imo

I think this makes sense. In the simple `broadcast` case we could say that "yeah no user would broadcast a staticcall", but here when there's a `startBroadcast` active, skipping staticcalls seems like a good UX upgrade, vs having to open/close for each staticcall? @mds1 is this convincing enough as an argument, that the UX deteriorates if we do not handle it?

Reply:
We can just collect the static calls and by default filter them out in the `script` command with a flag to include them. I don't mind either way (although I prefer less code in the inspector).

I also think there was some issue with `is_static` in REVM, at least some of the TODOs in REVM indicate so (cc @rakita)

Reply:
It seems it was only affecting `initialize_interp` from `Inspector`, not sure why I didn't fix it at that time. Made a pull request for it: https://github.com/bluealloy/revm/pull/109/files

Reply:
I don't feel too strongly and I'm ok with filtering out staticcalls by default, and I do agree it provides better UX. To summarize my rationale for preferring "all calls after startBroadcast are sent, static or not":

- Mainnet deploys for any non-trivial system are always terrifying (or maybe that's just me 😅), so a goal here should be to make these scripts as safe and readable as possible even if there's a small UX cost (cause the UX will still be 10x better than JS scripts)
- Therefore it should be very obvious which txs are actually being sent to the live network and which aren't. Reading/validating a deploy script seems simpler and less error prone if the reader doesn't need to know which calls are staticcalls that get filtered out vs. which actually get broadcast.
- Similarly, you might be calling a static method, but your interface definition left out `view`, and you end up sending a staticcall tx anyway without realizing because you assumed it'd get filtered out. So always sending all calls helps mitigate that class of errors.

Reply:
> Mainnet deploys for any non-trivial system are always terrifying (or maybe that's just me sweat_smile), so a goal here should be to make these scripts as safe and readable as possible even if there's a small UX cost (cause the UX will still be 10x better than JS scripts)

Maybe by default for each transaction that is sent, it is pretty printed to the console and the user must manually approve it at a `y/n` prompt. There could be a `--force` flag that would skip any user prompts. Then big deployments can be taken slowly and double checked at each step